### PR TITLE
Add test case for ORC statistics test

### LINKED ISSUE #8763: Statistics tests for ORC files written by GPU
When the GPU writes an ORC file, it's important to get the file, row group, and page statistics correct. We should have better tests, ideally including tests at larger scale than a few thousand rows typical for unit/integration tests, for verifying the statistics written to the ORC file are correct.

### PR DESCRIPTION
**Pull request by res-life**
closes #8763

Fix bugs in the Date/Timestamp generators in `bigDataGen`
Add test case for ORC statistics test

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
@revans2  Please have a look for the  `bigDataGen` fix part.
The `TimestampGenFunc` and `DoubleGenFunc` did not check the min/max value.

**Review by revans2 — COMMENTED** [review]
Generally it looks good.  Just the one nit about the value ranges in data gen.  I would be fine if we provided min/max ORC value ranges, or if we have a helper API to setup default ranges for ORC.  But I just don't think it should be the default everywhere.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

File: integration_tests/src/main/scala/org/apache/spark/sql/tests/datagen/bigDataGen.scala [hunk 1]
```diff
@@ -1321,12 +1328,35 @@ class TimestampGen(conf: ColumnConf,
   override def dataType: DataType = TimestampType
 }
 
+object BigDataGenConsts {
+  private val epoch = LocalDateTime.of(1970, 1, 1, 0, 0, 0, 0)
+  private val minDate = LocalDateTime.of(1, 1, 1, 0, 0, 0, 0)
+  private val maxDate = LocalDateTime.of(9999, 12, 31, 0, 0, 0, 0)
```
[the remark this answers is not in the document]
Comment (res-life) [reply integration_tests/src/main/scala/org/apache/spark/sql/tests/datagen/bigDataGen.scala:1331]: Updated.

**Comment by res-life** [comment]
I'll mark it as "Ready for review" after the premerge is passed.

**Review by revans2 — APPROVED** [review]

**Pull request merged**